### PR TITLE
Add releaseSegments interface for query node

### LINKED ISSUE #8144: Add releaseSegment interface for queryNode
#### Is your feature request related to a problem? Please describe:
Add releaseSegment interface for queryNode

### PR DESCRIPTION
**Pull request by bigsheeper**
Signed-off-by: bigsheeper <yihao.dai@zilliz.com>

Resolves: #8144

### CODE REVIEW

File: internal/querynode/impl.go [hunk 1]
```diff
@@ -385,24 +385,42 @@ func (node *QueryNode) ReleaseSegments(ctx context.Context, in *queryPb.ReleaseS
 		}
 		return status, err
 	}
-	status := &commonpb.Status{
-		ErrorCode: commonpb.ErrorCode_Success,
+	dct := &releaseSegmentsTask{
```
Comment (xige-16) [inline internal/querynode/impl.go:388]: if the segment to be released not exist in historical replica, the there is not need to create a task.

Comment (bigsheeper) [reply internal/querynode/impl.go:388]: fixed

File: internal/querynode/task.go [hunk 2]
```diff
@@ -600,3 +606,62 @@ func (r *releasePartitionsTask) Execute(ctx context.Context) error {
...
+		r.SetID(r.req.Base.MsgID)
+	}
+	return nil
+}
+
+func (r *releaseSegmentsTask) PreExecute(ctx context.Context) error {
+	return nil
+}
+
+func (r *releaseSegmentsTask) Execute(ctx context.Context) error {
```
Comment (xige-16) [inline internal/querynode/task.go:632]: if streaming and historical don't have any data of the collection after releasing segment, why not release collection meta and clear query collection?

Comment (bigsheeper) [reply internal/querynode/task.go:632]: fixed

**Pull request closed without merging**